src/js/dqa.jsx

```diff
@@ -80,20 +80,15 @@ var DQA = React.createClass({
     },
 
     _trimToLength: function(str, maxLengthInPixels) {
-        /*  Note: Always keep at least one word because column renderers of
-            fixed-data-table will always display at least part of the first word
-            (all words after the first are either displayed in full length or 
-            completely chopped off).
-        */
-        if (str.indexOf(' ') < 0) {
-            return str;
-        }
-
         var data = str.trim().split(' ');
-        var res = data[0];
-
-        for (var i=1; i<data.length; i++) {
-            if (this._getPixelLength(res + ' ' + data[i]) > maxLengthInPixels-20) {
+        var res = '';
+        maxLengthInPixels -= 20;
+
+        for (var i=0; i<data.length; i++) {
+            if ((this._getPixelLength(res + data[i]) > maxLengthInPixels) ||
```

Comment:
It is hard to understand what each of these expressions do in this conditional. Can you break it up in to variables and add a comment? Also I suggest updating the commit message to say `Fix table text truncation` since `overflow truncation` sounds paradoxical.